File: 1-two-sum/two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer,Integer>mpp=new HashMap<>();

        for(int i=0;i<nums.length;i++){
            int rem=target-nums[i];
            if(mpp.containsKey(rem)){
                return new int[]{i,mpp.get(rem)};
            }
            mpp.put(nums[i],i);
        }
        return new int[]{-1,-1};
    }
}

File: 1-two-sum/two-sum.py
class Solution:
    def twoSum(self, nums: list[int], target: int) -> list[int]:
        mp={}
        for x in range(0,len(nums)):
            rem=target-nums[x]
            if rem in mp:
                return [x,mp[rem]]
            mp[nums[x]]=x
        return [-1,-1]        

        